docs/API.md

```diff
@@ -0,0 +1,153 @@
+  __sf_util_set_default SPACEFISH_FOOBAR_SYMBOL "🍷 "
+  __sf_util_set_default SPACEFISH_FOOBAR_COLOR white
+
+  # ------------------------------------------------------------------------------
+  # Section
+  # ------------------------------------------------------------------------------
+
+  # If SPACEFISH_FOOBAR_SHOW is false, don't show the foobar section
+  [ $SPACEFISH_FOOBAR_SHOW = false ]; and return
+
+  # If the foobar command doesn't exist, don't show the foobar section
+  type -q foobar; or return
+
+  # Use `set -l` to define local variables to avoid populating
+  # the global namespace
+  set -l foobar_status
+
+  if test "$SOME_CONDITION" = "true"
+    set foobar_status (foobar baz)
+  else
+    set foobar_status (foobar foo)
+  end
+
+  # Display the foobar section
+  __sf_lib_section \
+  $SPACEFISH_FOOBAR_COLOR \
+  $SPACEFISH_FOOBAR_PREFIX \
+  $SPACEFISH_FOOBAR_SYMBOL \
+  $SPACEFISH_FOOBAR_SUFFIX
+end
+```
+
+## `SPACEFISH_VERSION`
+
+An environment variable that defines the version of currently running Spacefish prompt. Can be used for issue reporting or debugging purposes.
+
+Accessible to any program or script running in a current shell session.
+
+### Example
+
+```sh
+echo $SPACEFISH_VERSION
+#> 0.1.0
+```
+
+## `__sf_lib_section <color> [prefix] <content> [suffix]`
+
+This function prints out the prompt section prefixed with `prefix`, suffixed with `suffix` and `content` formatted to display in `color`. The **Bold** style is applied by default.
+
+`prefix`, `suffix` and `content` can contain `set_color` to set an additional foreground color, background color or other formatting styles. Read more about `set_color` in the [set_color - set the terminal color](https://fishshell.com/docs/current/commands.html#set_color) section of the Fish Shell documentation.
+
+If `SPACEFISH_PROMPT_PREFIXES_SHOW` is `false` or if the section is the first to appear in the prompt, then `prefix` will be omitted.
+
+If `SPACEFISH_PROMPT_SUFFIXES_SHOW` is `false`, then `suffix` will be omitted.
+
+Both `prefix` and `suffix` are optional. They are equal to empty strings by default.
+
+### Arguments
+
+1. `color` _Required_ — The color used when displaying the `content`. Can be any of the valid [basic colors](https://fishshell.com/docs/current/commands.html#set_color) or can be any valid RGB hex code.
+2. `prefix` _Optional_ — The prefix shown before `content`. Usually, it's the value of `SPACEFISH_*_PREFIX`.
+3. `content` _Required_ — The content of the section. Can be any valid value or the result of command execution.
+4. `suffix` _Optional_ — The suffix shown after `content`. Usually, it's the value of `SPACEFISH_*_SUFFIX`.
+
+### Example
+
+```sh
+# Display the prompt section with a prefix and suffix
+# Backslash is used to escape the line endings
+__sf_lib_section \
+$SPACEFISH_SECTION_COLOR \
+$SPACEFISH_SECTION_PREFIX \
+"$SPACEFISH_SECTION_SYMBOL$section_content" \
```

Comment:
The double-quotes here aren't actually necessary.
Should `$SPACEFISH_SECTION_SYMBOL$section_content` or `"$SPACEFISH_SECTION_SYMBOL$section_content"` be used?

Reply:
Absolutely true!
I'll strip those away. 😄

Reply:
🤕 
So it appears that if one of the two values is null, the concatenated expression is null. Seems like a strange fish quirk which doesn't happen if they're wrapped in quotes.

Unfortunately bumped into this when the git section wouldn't print because I had an empty git status.

Just a heads up!